File: submissions/kasonchan/submission3/compositions/B.scala
/**
 * File: B.scala
 * Created Date: June 6, 2014
 * Status: Completed
 * Programmer: Ka-Son Chan kason.chan@ttu.edu
 * Description: This file contains the task B operations of the task graph.
 */

import java.util.Scanner
import java.io._

object B {
  def main(args: Array[String]) = {
    taskB(args(0), args(1))
  }
  
  def subtract(x: Int)(y: Int) = x - y

  def taskB(intputFileName: String, outputFileName: String) {
    // Read input file
    val scanner = new Scanner(new File(intputFileName))
    val data = scanner.nextInt()
    scanner.close()

    // Perform subtraction
    val result = subtract(data)(1)

    // Write result to output file
    val fileWriter = new PrintWriter(outputFileName)
    fileWriter.write((result).toString)
    fileWriter.write("\n")
    fileWriter.close()
  }
}

File: submissions/kasonchan/submission3/compositions/B.c
/**
 * File: B.c
 * Created Date: June 12, 2014
 * Status: Completed
 * Programmer: Ka-Son Chan kason.chan@ttu.edu
 * Description: This file contains the task B operations of the task graph.
 */

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

// Function declaration
int subtract(int x, int y);
void taskB(char * inputFileName, char * outputFileName);

int main(int argc, char * argv[]) {
  // Check the argument number
  if (argc != 3) {
    printf("Error - invalid number of arguments\n");
  }

  taskB(argv[1], argv[2]);

  return 0;
}

// Function definitions
int subtract(int x, int y) {
  return x - y;
} 

void taskB(char * inputFileName, char * outputFileName) {
  int result;
  FILE * fpInput;
  FILE * fdOutput;
  fpInput = fopen(inputFileName, "r");  
  fdOutput = fopen(outputFileName, "w");

  // Read from input file
  fscanf (fpInput, "%d", &result); 

  // Perform subtraction
  result = subtract(result, 1);

  // Write output to file
  fprintf(fdOutput, "%d\n", result);
  
  // Close files
  fclose(fpInput);
  fclose(fdOutput);
}
